### PR TITLE
vc: vn-agent: test for /containerLogs, /run api proxy

### PR DESCRIPTION
**Pull request by zhuangqh**
the principle job is to test the naming translation of each api.
Signed-off-by: jerryzhuang <zhuangqhc@gmail.com>

### CODE REVIEW

File: incubator/virtualcluster/pkg/vn-agent/server/server_test.go [hunk 1]
```diff
@@ -413,9 +420,13 @@ func getPodName(name, namespace string) string {
 	return name + "_" + namespace
 }
 
+func getTenantNamespace(tenantName, namespace string) string {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/vn-agent/server/server_test.go]: How about getTenantNamespace -> getEffectiveNamespace, also add a comment to this method.

File: incubator/virtualcluster/pkg/vn-agent/server/server_test.go [hunk 2]
```diff
@@ -444,3 +455,167 @@ func TestServeLogs(t *testing.T) {
...
+	if result != output {
+		t.Errorf("expected %s, got %s", output, result)
+	}
+}
+
+func TestServeRunInContainerWithUID(t *testing.T) {
+	// TODO: support UID naming translate
+}
+
+func setPodByNameFunc(fv *serverTestFramework, namespace, pod, container string) {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/vn-agent/server/server_test.go]: Just curious, why do you extract this method out? the same for setGetContainerLogsFunc? 
(you did not do this for runFunc in TestServeRunInContainer)

Comment (zhuangqh) [reply incubator/virtualcluster/pkg/vn-agent/server/server_test.go]: > Just curious, why do you extract this method out? the same for setGetContainerLogsFunc?
> (you did not do this for runFunc in TestServeRunInContainer)

there is a validation in the kubelet server, it gets the Pod spec from `GetPodByName` interface and check whether there exists the correspond container. We inject this info from `setPodByNameFunc` this method. However, there is no this validation in the `/run` api.  It's strange...

File: incubator/virtualcluster/pkg/vn-agent/server/server_test.go (hunk 1 above)
Comment (zhuangqh) [reply incubator/virtualcluster/pkg/vn-agent/server/server_test.go]: updated

**Pull request merged**